arraycontext/container/arithmetic.py

```diff
@@ -343,6 +343,17 @@ def {fname}(arg1):
                                 else:
                                     raise ValueError(msg)""")
                     gen(f"return cls({zip_init_args})")
+                if _cls_has_array_context_attr:
```

Comment:
This should get its own flag, defaulted to the same value as `bcast_number`.

Reply:
Done, see here: 93013c9.